app/scripts/migrations/120.2.ts

```diff
@@ -0,0 +1,48 @@
+import { cloneDeep } from 'lodash';
+import { hasProperty, isObject } from '@metamask/utils';
+
+type VersionedData = {
+  meta: { version: number };
+  data: Record<string, unknown>;
+};
+
+export const version = 120.2;
+
+/**
+ * This migration removes any dangling instances of SelectedNetworkController.perDomainNetwork
+ *
+ * @param originalVersionedData - Versioned MetaMask extension state, exactly what we persist to dist.
+ * @param originalVersionedData.meta - State metadata.
+ * @param originalVersionedData.meta.version - The current state version.
+ * @param originalVersionedData.data - The persisted MetaMask state, keyed by controller.
+ * @returns Updated versioned MetaMask extension state.
+ */
+export async function migrate(
+  originalVersionedData: VersionedData,
+): Promise<VersionedData> {
+  const versionedData = cloneDeep(originalVersionedData);
+  versionedData.meta.version = version;
+  transformState(versionedData.data);
+  return versionedData;
+}
+
+function transformState(state: Record<string, unknown>) {
+  if (!hasProperty(state, 'SelectedNetworkController')) {
+    return state;
+  }
+
+  if (!isObject(state.SelectedNetworkController)) {
+    global.sentry?.captureException?.(
```

Comment:
Hmm, do we think this would be actionable? It certainly would be unexpected, but we're solving the problem in this same migration.

Typically with state validation we want to do one of two things with unexpected state:
* Capture a Sentry error and return state unmodified, for us to fix in a follow up release, or
* Log an error to the console and fix it. We might not learn it happened, but it's there in the console logs if we're investigating a related report.

If we're fixing the problem here, maybe we should log an error rather than capture an exception.

Reply:
Yeah agreed not really actionable. Will switch it to a `console.error`. We don't really have a pattern established for logging errors from migrations.

```
  console.error(
      `Unexpected state encountered during migration: state.SelectedNetworkController is type: ${typeof state.SelectedNetworkController}`,
    );
```
?

Reply:
Sounds reasonable to me!